opal/mca/btl/tcp/btl_tcp_proc.c

```diff
@@ -355,6 +355,10 @@ static mca_btl_tcp_interface_t** mca_btl_tcp_retrieve_local_interfaces(void)
             if (6 == mca_btl_tcp_component.tcp_disable_family) {
                 continue;
             }
+            if (opal_net_islinklocal(&local_addr)) {
```

Comment:
Per the comment below, if opal_net_islinklocal() grows to understand IPv4, we might as well move this check outside of the switch/case block and do it for both IPv4 and IPv6.